File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        set<ListNode*> st;
        while(headA!=nullptr or headB!=nullptr){
            if(headA==headB){
                return headA;
            }
            if(headA!=nullptr and st.count(headA)){
                return headA;
            }
            if(headB!=nullptr and st.count(headB)){
                return headB;
            }
            if(headA){
                st.insert(headA);
                headA=headA->next;
            }
            if(headB){
                  st.insert(headB);
                headB=headB->next;
            }
        }
        return nullptr;
    }
};

File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function getIntersectionNode(headA: ListNode | null, headB: ListNode | null): ListNode | null {
    let mp : Map<(ListNode|null),number> = new Map();
    while(headA!=null){
        mp.set(headA,1);
        headA=headA.next;
    }
    while(headB!=null){
        if(mp.get(headB)){
            return headB;
        }
         headB=headB.next;
    }
    return null;
};